src/index.ts

```diff
@@ -82,11 +82,11 @@ class Client<RequestType, ResponseType> {
     );
   }
 
-  listResource<T>(uri: string): Promise<Array<T>> {
-    const request = this.requestFactory.createRequest(uri);
-    return this.requestPerformer
-      .performRequest(request)
-      .then(this.responseParser.parseResponse);
+  listResource<T>(
+    uri: string,
+    params?: Record<string, any> | null | undefined,
```

Comment:
Record means anything Map like?

Reply:
Yes, sort of. `Record<Keys, Type>` is an elegant way of typing objects.